mysql-test/suite/innodb/t/alter_copy_bulk.test

```diff
@@ -42,3 +42,22 @@ INSERT INTO t1 SELECT seq, seq * 2 FROM seq_3_to_65536;
 # Alter should use temporary file for sorting
 ALTER TABLE t1 ADD INDEX(f2);
 DROP TABLE t1;
+
+--echo #
+--echo # MDEV-34756 Validation of new foreign key skipped
+--echo #            if innodb_alter_copy_bulk=ON
+--echo #
+CREATE TABLE t1(f1 INT NOT NULL PRIMARY KEY,
+		f2 INT NOT NULL)ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT NOT NULL PRIMARY KEY,
+		f2 INT NOT NULL)ENGINE=InnoDB;
+INSERT INTO t2 VALUES (1, 1);
+--replace_regex /#sql-alter-[0-9a-f-]*/#sql-alter/
+--error ER_NO_REFERENCED_ROW_2
+ALTER TABLE t2 ADD CONSTRAINT FOREIGN KEY(f2) REFERENCES t1(f1);
+SET session FOREIGN_KEY_CHECKS=0;
+INSERT INTO t1 VALUES(1, 1);
+ALTER TABLE t2 ADD CONSTRAINT FOREIGN KEY(f2) REFERENCES t1(f1);
+SET session FOREIGN_KEY_CHECKS=1;
+ALTER TABLE t1 FORCE;
```

Comment:
I would suggest `--enable_info` and `--disable_info` around this, to demonstrate that the `ALGORITHM=COPY` code path is being used here. It should report "1 row(s) affected", while `ALGORITHM=INPLACE` should always report "0 row(s) affected".

Please add an `ALTER TABLE` right before the `INSERT`, to show that no error will be reported for the violation. I would expect that one to use the `ALGORITHM=INPLACE` code path, which should be indicated by the `--enable_info` output.

I’d also suggest writing
```sql
SET STATEMENT foreign_key_checks=0 FOR
ALTER TABLE t2 ADD CONSTRAINT FOREIGN KEY(f2) REFERENCES t1(f1);
```
so that there is no need to "permanently" change the session variable.

Reply:
OK. Fixed the issue.